vyxal/transpile.py

```diff
@@ -0,0 +1,291 @@
+            )
+    if isinstance(struct, structure.Lambda):
+        id_ = secrets.token_hex(16)
+        # The lambda id used to be based on time.time() until
+        # I realised just how useless that was, because the calls to
+        # time.time() happened within only a few milliseconds of each
+        # other, meaning int(time.time()) would return the exact same
+        # lambda name for multiple lambdas.
+
+        return (
+            indent_str(
+                f"def _lambda_{id_}(parameters, arity, self, ctx):",
+                indent,
+            )
+            + indent_str(f"this = _lambda_{id_}", indent + 1)
+            + indent_str("overloaded_arity = False", indent + 1)
+            + indent_str(
+                'if "arity_overload" in dir(self): '
+                + "overloaded_arity = self.arity_overload",
+                indent + 1,
+            )
+            + indent_str(
+                f"if arity and arity != {struct.branches[0]}: "
+                + "stack = pop(parameters, arity, ctx))",
+                indent + 1,
+            )
+            + indent_str(
+                "elif overloaded_arity: stack = pop(parameters, arity"
+                + ", ctx",
+                indent + 1,
+            )
+            + indent_str(
+                f"else: stack = pop(parameters, {struct.branches[0]}[0]"
+                + ", ctx)",
+                indent + 1,
+            )
+            + indent_str("ctx.context_values.append(stack[::])", indent + 1)
+            + indent_str("ctx.input_level += 1", indent + 1)
+            + indent_str(
+                "ctx.input_values[ctx.input_level] = [stack[::], 0]",
+                indent + 1,
+            )
+            + transpile_ast(struct.branches[1], indent + 1)
+            + indent_str("ret = pop(stack, ctx=ctx)", indent + 1)
+            + indent_str("ctx.context_values.pop()", indent + 1)
+            + indent_str("ctx.input_level -= 1", indent + 1)
+            + indent_str("return ret", indent + 1)
+            + indent_str(f"stack.append(_lambda_{id_})", indent)
+        )
+
+    if isinstance(struct, structure.FunctionReference):
+        return indent_str(f"stack.append(FN_{struct.branches[0]})", indent)
+
+    if isinstance(struct, structure.ListLiteral):
+        # We have to manually build this because we don't know how
+        # many list items there will be.
+
+        temp = indent_str("temporary_list = []", indent)
+        for x in struct.items:
+            temp += (
+                indent_str("def list_item(s, ctx):", indent)
+                + indent_str("stack = s[::]", indent + 1)
+                + transpile_ast(x, indent + 1)
+                + indent_str("return pop(stack, ctx=ctx)", indent + 1)
+                + indent_str(
+                    "temporary_list.append(list_item(stack, ctx)", indent
+                )
+            )
+
+        temp += indent_str("stack.append(temp_list[::]", indent)
```

Comment:
This is a *big* problem - which are we using, `temporary_list` or `temp_list`? Because this is going to be bad. Also how do we avoid name conflicts with that, and list_item?

Reply:
I say shorter names are better.

Reply:
This isn't an issue because function scoping.

Reply:
Also, `temp_list`.